docs/architecture/deep-dives/passkeys/implementations/relying-party/prf.md

```diff
@@ -0,0 +1,101 @@
+necessary for passkeys to be used for decryption as well.
+
+## What is PRF?
+
+At the most abstract level, account encryption and decryption requires that the user's symmetric key
+be encrypted ("protected") with another key that is available only to the client. For master
+password-based decryption, this is done with the master key, derived through a key derivation
+function from the master password.
+
+In order to use a passkey for account decryption on login, we need such a key available on the
+passkey, with which we can encrypt and decrypt the user's symmetric key. Fortunately, the `prf`
+extension to the WebAuthn spec provides exactly this functionality.
+
+The `prf` extension, which is built upon the `hmac-secret` CTAP2 extension, provides the
+specification by which credential assertion can return the output from a pseudo-random function
+(hence "PRF") that is unique to each credential on the authenticator. The Bitwarden relying party
+can leverage this to obtain a unique, cryptographically-secure key, with which it can encrypt and
+decrypt the user's symmetric key - analogous to the derived master key when using a master password
+to log in.
+
+## How does Bitwarden use the PRF key?
+
+### During passkey registration
+
+When a passkey is registered for use with Bitwarden, we interrogate the attestation response to see
+if PRF is supported. If so, we allow the user to choose to enable encryption.
+
+![Registering a passkey with PRF](image.png)
+
+If the user chooses to enable encryption, we request an assertion from the authenticator, which will
+provide the key. (This is why you may be prompted multiple times to provide user verification when
+registering a PRF-capable passkey.)
+
+The 32-byte key returned from the authenticator is then stretched to a 64-byte two-part key using
+HKDF. The first 32 bytes will serve as the AES-256 encryption key and the second 32 bytes will serve
+as a message authentication code (MAC).
+
+Once we have this encryption key, we can encrypt the user's symmetric key as follows:
+
+    1. Create an RSA public/private key pair, specific to this credential.
+    2. Encrypt the RSA private key with the encryption key.
+    3. Encrypt the user's symmetric key with the RSA public key.
+
+The PRF-encrypted private key, the public key-encrypted encryption key, and the RSA public key are
+persisted to the server and associated with the registered passkey. Note that the PRF key does
+**not** leave the client at any point, to ensure zero-knowledge encryption.
+
+:::tip Why an RSA key pair?
+
+Put simply: key rotation. In order to support
+[encryption key rotation](https://bitwarden.com/help/account-encryption-key/#rotate-your-encryption-key),
+a user must be able to re-encrypt the symmetric key for all registered passkeys without gaining
+access to the PRF keys for each registered credential. The use of a public/private keypair allows a
```

Comment:
> a user **must** ... re-encrypt the symmetric key ... without gaining access to ... each registered credential.

nit: "must" is a strong word. There's nothing that inherently dictates that. This is more about a choice we've made to simplify the user experience.

suggestion (with room for improvement):
In order to support [encryption key rotation](https://bitwarden.com/help/account-encryption-key/#rotate-your-encryption-key), a user must be able to re-encrypt the symmetric key for all registered passkeys. If the PRF key was used to directly encrypt the user's symmetric key then this operation would require gaining access to the PRF keys for each registered credential. The use of a public/private keypair allows the user to re-encrypt the new symmetric key with the public key, with the assurance that only the holder of the private key can decrypt it, and without having to provide access to the PRF key for each registered credential.

Reply:
I think your suggestion sounds great!  I've updated the copy in [4ab27ff (#307)](https://github.com/bitwarden/contributing-docs/pull/307/commits/4ab27ff50f7a78620ea68d6f99691c481f24c263).